rpc/utils.go

```diff
@@ -218,29 +218,27 @@ func newSubscriptionId() (string, error) {
 // SupportedModules returns the collection of API's that the RPC server offers
 // on which the given client connects.
 func SupportedModules(client Client) (map[string]string, error) {
-	req := map[string]interface{}{
-		"id":     1,
-		"method": "rpc_modules",
+	req := JSONRequest{
+		Id:      new(int64),
+		Version: "2.0",
+		Method:  "rpc_modules",
 	}
-
 	if err := client.Send(req); err != nil {
 		return nil, err
 	}
 
-	var response map[string]interface{}
-	if err := client.Recv(&response); err != nil {
+	response := new(JSONSuccessResponse)
```

Comment:
1. Probably old legacy code before we had custom JSON message types.
2. It doesn't matter how you declare it. Go does escape analysis. If it deems that data might leave the function, it's placed on the heap, otherwise stack automatically. You have no control over it at all.